joomla/joomla-cms
# [5.0] Introduce a CLI UI for creating a public folder

Pull Request for Issue # .

### Summary of Changes

- Added 1 new command on the main `joomla.php` CLI:

```
site
  site:create-public-folder       Create a public folder
```

- Added one more step on the CLI installation (will do nothing if nothing inserted):

```
 Please enter the absolute path to the public folder:
```

- The code is requiring https://github.com/joomla/joomla-cms/pull/41570


### 3rd Part Devs impact:
- If you have no direct Entry Point (similar to index.php) you're not affected. But since there might be legitimate cases (backup script, auditing scripts, etc) that boot their own app instead of Joomla there's this idea to make life easier:

> Hmm, now that I'm thinking about 3rd PD maybe a simple XML line like the following is all that's needed. We parse all the admin component manifests and symlink the entry points

```
<exposed-entrypoints>
  <path>administrator/components/com_foo/dangerous-entry.php</path>
  <path>administrator/components/com_foo/another-dangerous-entry.php</path>
</exposed-entrypoints>
```

- Another possible conflict might be that most folders in the public folder would get by default an htaccess file with `<files>deny from all</files>`, meaning NO MORE PHP FILES inside `media`, `images` and any other media manager storage folder. If your extension is placing PHP files in these folders you're doing it very wrong, please consider moving them to their appropriate folder.

- Have some input? Please share it..

### Testing Instructions

#### Testing a *NEW* installation 

- Apply this PR, ie check out this PR
- go to the installation folder in your root Joomla installation: `cd installation ` 
- run the command: `php joomla.php install`
- insert the FULL path for the public folder, ie 
<img width="727" alt="Screenshot 2023-08-25 at 16 35 02" src="https://github.com/joomla/joomla-cms/assets/3889375/7af0350d-6da5-41df-8bc5-535d60e94f34">

- Switch your server to the new directory.

#### Testing an *EXISTING* installation

- Download the package for the GitHub or check out this PR
- go to the `cli` folder in your root Joomla installation: `cd cli` 
- run the command: `php joomla.php site:create-public-folder`
- insert the FULL path for the public folder, ie <img width="727" alt="Screenshot 2023-08-25 at 16 35 02" src="https://github.com/joomla/joomla-cms/assets/3889375/7af0350d-6da5-41df-8bc5-535d60e94f34">

- Switch your server to the new directory.


### Actual result BEFORE applying this Pull Request


### Expected result AFTER applying this Pull Request


### Link to documentations
WIP


@SniperSister @HLeithner @wilsonge thoughts?

@Hackwar I would appreciated some fixes here and some help for the installation CLI
